source/fundamentals/monitoring.txt

```diff
@@ -0,0 +1,396 @@
+==========
+Monitoring
+==========
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecols
+
+Overview
+--------
+
+This guide shows you how to monitor topology events in a MongoDB instance,
+replica set, or sharded cluster. Topology events, also known as Server
+Discovery and Monitoring (SDAM) events, are created by the driver when there is
+a change in the state of the instance or cluster that you are connected to.
+For example, when a new connection is established or a new primary is elected,
+an event is created.
+
+Read this guide if you need to record topology changes in your application or
+want to explore the information provided in these events.
+
+Event Descriptions
+------------------
+
+The following table lists all the SDAM events that are made available to your
+application by the driver:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Event Name
+     - Description
+
+   * - ``serverOpening``
+     - Created when a connection to an instance is established.
+
+   * - ``serverClosed``
+     - Created when a connection to an instance is closed.
+
+   * - ``serverDescriptionChanged``
+     - Created when an instance state changes (such as from secondary to
+       primary).
+
+   * - ``topologyOpening``
+     - Created before any instance connections are attempted.
+
+   * - ``topologyClosed``
+     - Created after all instance connections in the topology are closed.
+
+   * - ``topologyDescriptionChanged``
+     - Created when the topology changes, such as a new primary being
+       elected or a **mongos** proxy disconnecting.
+
+       *Not applicable to a single instance.*
+
+   * - ``serverHeartbeatStarted``
+     - Created prior to issuing an ``isMaster`` command to a MongoDB instance.
+
+       *Not applicable to a single instance.*
+
+   * - ``serverHeartbeatSucceeded``
+     - Created when a successful response to the ``isMaster`` command is
+       returned by a MongoDB instance.
+
+       *Not applicable to a single instance.*
+
+   * - ``serverHeartbeatFailed``
+     - Created when an ``isMaster`` command issued to a specific MongoDB
+       instance fails to return a successful response
+
+       *Not applicable to a single instance.*
+
+
+Event Subscription Example
+--------------------------
+
+You can access SDAM events through the driver by subscribing to the event
+in your application. The following example demonstrates connecting to a
+replica set and subscribing to all the SDAM events created by the MongoDB
+deployment:
+
+.. code-block:: js
```

Comment:
Both this example and the first example on the logging page are full programs, runnable aside from the uri. I would suggest moving both of these into separate files in the code-samples directory and literalincluding them (you can deindent to make them look just like a code-block) instead to ease testing when we eventually get to that point. Not a requirement, but a suggestion to make our lives easier in the future.

Reply:
Sure, although I had noticed them in your CRUD guide, I was not certain we were doing this for all code snippets. I'll have to review what I have for the Authentication section since I did not include the code there either.